symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,138 @@
+package com.symphony.bdk.core.service.datafeed.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.V4Event;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Slf4j
+public class DatafeedServiceV1 extends AbstractDatafeedService {
+
+    private static final String DATAFEED_ID_FILE = "datafeed.id";
+    private final AtomicBoolean started = new AtomicBoolean();
+    private String datafeedId;
+
+
+    public DatafeedServiceV1(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, podClient, authSession, config);
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws Throwable {
+        this.datafeedId = this.retrieveDatafeedIdFromDisk();
+
+        if (this.datafeedId == null) {
+            this.datafeedId = this.createDatafeedAndSaveToDisk();
+        }
+
+        log.debug("Start reading events from datafeed {}", datafeedId);
+        do {
+            this.started.set(true);
```

Comment:
if I do a start from a thread, and a stop from another thread that is executed before this line then my datafeed won't stop maybe we need some synchronization here

Reply:
If I'm not mistaken, the `AtomicBoolean` is thread-safe so the datafeed will stop by a stop from another thread.